src/pages/signup.astro

```diff
@@ -56,9 +60,18 @@ import '../styles/styles.css';
         const email = emailInput.value;
         const password = passwordInput.value;
 
-        console.log(name, email, password);
-
+        try {
+          await createUserWithEmailAndPassword(auth, email, password);
```

Comment:
You need to have an additional server-side component for authentication. The Astro documentation on Firebase auth https://docs.astro.build/en/guides/backend/google-firebase/#creating-auth-server-endpoints has explicit server-side _endpoints_, but in this case we can move that to Astro's built-in Actions feature.